common/component/azure/servicebus/client.go

```diff
@@ -38,14 +38,16 @@ type Client struct {
 	metadata    *Metadata
 	lock        *sync.RWMutex
 	senders     map[string]*servicebus.Sender
+	topicCache  map[string]bool
 }
 
 // NewClient creates a new Client object.
 func NewClient(metadata *Metadata, rawMetadata map[string]string) (*Client, error) {
 	client := &Client{
-		metadata: metadata,
-		lock:     &sync.RWMutex{},
-		senders:  make(map[string]*servicebus.Sender),
+		metadata:   metadata,
+		lock:       &sync.RWMutex{},
+		senders:    make(map[string]*servicebus.Sender),
+		topicCache: make(map[string]bool),
```

Comment:
Use `map[string]struct{}` since the value doesn't matter. You will then check if the value is in the cache by doing:

`if _, ok := topicCache[key]; ok`